app/ui/browser-modern/sagas/page-sagas.js

```diff
@@ -77,6 +79,9 @@ export function* destroyPageSession({ id, options = {} }) {
 
   if (!options.withoutUI) {
     const currentPageCount = yield select(PageSelectors.getPageCount);
+    const sessionId = yield apply(userAgentHttpClient, userAgentHttpClient.waitForSession, [page]);
+
+    yield put(PageActions.saveRestorablePage(page.id, sessionId));
```

Comment:
We should probably reconsider some of this -- this is the first time session IDs are exposed to the frontend (we abstract the page id <-> session id relationship in UserAgentHttpClient), hence just now running into this problem